### ISSUE TITLE
Scaling a model produces gcode with empty layers and build platform fails to raise to meet nozzle

### ISSUE THREAD

**Issue by castellotti** [daid/LegacyCura#993]
I'm having repeated trouble with certain prints beginning without the build platform being fully raised into position. Specifically, I have one gcode file generated by Cura 14.09 (with matching firmware on an Ultimaker 2) which exhibits this issue with perfect reproducibility. After selecting "Print" and then the file, the head will heat up, the build platform will raise to ~3cm below the nozzle, some initial filament will begin to extrude, and then when the head moves to the appropriate X and Y to begin the print, the build platform fails to continue to raise to meet the nozzle.

With other files I can reliably get a correct print started. Turning the machine off and on seems to have no effect, as does attempting to reposition the model inside Cura and re-saving to the SD card.

I'm happy to upload an example gcode file (with matching STL) if that would be useful.

Cheers

**Comment by Dim3nsioneer** [comment]
Yes, if you please could post here just the first 10-20 lines of your gcode file, please...

**Comment by castellotti** [comment]
Certainly, here you go:

;FLAVOR:UltiGCode
;TIME:3672  
;MATERIAL:6583  
;MATERIAL2:0         

;Layer count: 268
;LAYER:0
M107
;LAYER:1
M106 S127
;LAYER:2
M106 S255
;LAYER:3
;LAYER:4
;LAYER:5
;LAYER:6
;LAYER:7
;LAYER:8
;LAYER:9
;LAYER:10

**Comment by castellotti** [comment]
More information.

The STL file is available online here:

http://www.thingiverse.com/download:699662

It needed to be scaled to 1.5 (150%) in all directions in order to support 2.85mm spools. Printing at a smaller size seemed to go okay, scaling is what seems lead to those empty layer entries.

Here's the first several lines of gcode generated from same STL without scaling:

;FLAVOR:UltiGCode
;TIME:4492  
;MATERIAL:9058  
;MATERIAL2:0         

;Layer count: 178
;LAYER:0
M107
G0 F9000 X55.801 Y105.401 Z0.300
;TYPE:SKIRT
G1 F1200 X63.900 Y105.400 E0.97188
G1 X166.096 Y105.400 E13.23540
G1 X166.096 Y105.401 E13.23552
G1 X174.200 Y105.400 E14.20800
G1 X174.200 Y119.600 E15.91200
G1 X69.100 Y119.600 E28.52400
G1 X69.100 Y119.610 E28.52520
G1 X65.934 Y119.600 E28.90512
G1 X63.900 Y119.600 E29.14920
G1 X63.900 Y119.594 E29.14992
G1 X55.802 Y119.568 E30.12169
G1 X55.801 Y105.401 E31.82173
G10
G0 F9000 X60.001 Y109.601
;TYPE:WALL-INNER
G11
G1 F1200 X64.900 Y109.600 E32.40961
G1 X64.900 Y110.575 E32.52661
G1 X68.100 Y110.575 E32.91061

**Comment by castellotti** [comment]
Issue is also exhibited after rolling back both software and UM2 firmware to 14.07

**Comment by Dim3nsioneer** [comment]
There is a problem with the model. Check the x-ray view. You'll see a red plane... this might be the reason...

**Comment by castellotti** [comment]
Summary after further testing:
- Opening this STL in Cura as-is and generating gcode produces a successful print
- Opening this STL in Cura and then scaling to 1.5x generates gcode with empty layers and fails to raise build platform during print
- Opening this STL in Slic3r 1.1.7, scaling to 1.5x, exporting as a new STL, and opening in Cura generates gcode which produces a successful print
- Opening this STL in PreForm 1.7 (for Form 1 printer) generates a warning "There were issues loading the following models .. Integrity issues. Would you like PreForm to attempt repair?" Automatic repair is successful and permits printing both original and a scaled model

So in summary, even if there may be some issue with this particular STL, Cura can still slice it but fails when attempting to scale first. Various other software is able to compensate. Shouldn't that be treated as a bug with scaling and/or parsing certain STLs in Cura?

**Comment by daid** [comment]
Cura (as well as most other slicing software) does a best effort at fixing models. But no software slices all broken models. PreForm uses other repair mechanisms then Cura, and thus provides different results. For some models it will get better results, for some it will get worse results. There simply isn't a universal "fix everything" right now.
